ui/src/lib/yjs/useYjsStore.ts

```diff
@@ -28,67 +28,77 @@ export default ({
-        yWebSocketRef.current = new WebsocketProvider(
-          websocket,
-          workflowId,
-          yDoc,
-          { params: { token: websocketToken } },
-        );
-      }
+  // Initialize store and connect
+  const [{ yWorkflows, currentUserClientId }] = useState(() => {
+    const doc = new Y.Doc();
+    const yWorkflows = doc.getArray<YWorkflow>("workflows");
 
-      const yWorkflows = yDoc.getArray<YWorkflow>("workflows");
+    // Initialize with default workflow if empty
+    if (yWorkflows.length === 0) {
       const yWorkflow = yWorkflowBuilder("main", "Main Workflow");
       yWorkflows.push([yWorkflow]);
+    }
 
-      const currentUserClientId = yDoc.clientID;
-
-      // NOTE: any changes to the yDoc should be wrapped in a transact
-      const undoTrackerActionWrapper = (callback: () => void) =>
-        yDoc.transact(callback, currentUserClientId);
-
-      return { yWorkflows, currentUserClientId, undoTrackerActionWrapper };
-    });
+    return {
+      yWorkflows,
+      currentUserClientId: doc.clientID,
+    };
+  });
 
   useEffect(() => {
-    if (yWorkflows) {
-      const manager = new Y.UndoManager(yWorkflows, {
-        trackedOrigins: new Set([currentUserClientId]), // Only track local changes
-        captureTimeout: 200, // default is 500. 200ms is a good balance between performance and user experience
+    const { websocket: websocketUrl, websocketToken } = config();
+    if (!websocketUrl || !websocketToken || !workflowId || !currentProject?.id)
+      return;
+
+    const doc = managerRef.current?.getDoc() || yWorkflows.doc;
+
+    // Create and setup socket manager
+    if (!doc) return;
+    const manager = new SocketYjsManager(doc);
+    manager
+      .setupSocket({
+        url: websocketUrl,
+        roomId: workflowId,
+        projectId: currentProject.id,
+        accessTokenProvider: async () => websocketToken,
+      })
+      .catch((error) => {
+        toast({
+          title: t("Connection Error"),
+          description: error.message,
+          variant: "destructive",
+        });
       });
-      setUndoManager(manager);
-
-      return () => {
-        manager.destroy(); // Clean up UndoManager on component unmount
-      };
-    }
-  }, [yWorkflows, currentUserClientId]);
-
-  const handleWorkflowUndo = useCallback(() => {
-    if (undoManager?.undoStack && undoManager.undoStack.length > 0) {
-      undoManager?.undo();
-    }
-  }, [undoManager]);
-
-  const handleWorkflowRedo = useCallback(() => {
-    if (undoManager?.redoStack && undoManager.redoStack.length > 0) {
-      undoManager?.redo();
-    }
-  }, [undoManager]);
 
+    // Setup undo manager
+    const undoMngr = new Y.UndoManager(yWorkflows, {
+      trackedOrigins: new Set([currentUserClientId]),
+      captureTimeout: 200,
+    });
+    setUndoManager(undoMngr);
+
+    managerRef.current = manager;
+
+    return () => {
+      manager.destroy();
+      undoMngr.destroy();
+      managerRef.current = null;
```

Comment:
Is `manager.destroy()` necessary if we are setting the managerRef.current to null anyways?